src/shared/varlink.c

```diff
@@ -2267,6 +2268,58 @@ int varlink_callb_ap(
         return varlink_call_full(v, method, parameters, ret_parameters, ret_error_id, ret_flags);
 }
 
+int varlink_call_and_log(
+                Varlink *v,
+                const char *method,
+                JsonVariant *parameters,
+                JsonVariant **ret_parameters,
+                const char **ret_error_id) {
+
+        JsonVariant *reply = NULL;
+        const char *error_id = NULL;
+        int r;
+
+        assert_return(v, -EINVAL);
+        assert_return(method, -EINVAL);
+
+        r = varlink_call(v, method, parameters, &reply, &error_id);
+        if (r < 0)
+                return log_error_errno(r, "Failed to issue %s() varlink call: %m", method);
+        if (error_id)
+                return log_error_errno(varlink_error_to_errno(error_id, reply),
+                                         "Failed to issue %s() varlink call: %s", method, error_id);
+
+        if (ret_parameters)
+                *ret_parameters = TAKE_PTR(reply);
+        if (ret_error_id)
+                *ret_error_id = error_id;
```

Comment:
mmh this will never return anything but NULL, no? If it's defined, the function returns earlier?

Reply:
hmm, yes, indeed.